src/WinRT.Runtime/Projections/EventHandler.cs

```diff
@@ -200,11 +207,16 @@ public NativeDelegateWrapper(ObjectReference<global::WinRT.Interop.IDelegateVftb
             public unsafe void Invoke(object sender, T args)
             {
 #if NET
-                bool useDynamicInvoke = EventHandlerMethods<T>._Invoke == null && RuntimeFeature.IsDynamicCodeCompiled;
-#else
-                bool useDynamicInvoke = EventHandlerMethods<T>._Invoke == null;
+                // Standalone path for NAOT to ensure the linker trims code as expected
+                if (!RuntimeFeature.IsDynamicCodeCompiled)
+                {
+                    EventHandlerMethods<T>._Invoke(_nativeDelegate, sender, args);
+
+                    return;
+                }
 #endif
-                if (!useDynamicInvoke)
+
+                if (EventHandlerMethods<T>._Invoke == null)

```

Comment:
```suggestion
                if (EventHandlerMethods<T>._Invoke != null)

```

Reply:
Oops, good catch! Weird though that no test caught this 😅